<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="namespace" default="taluy" />

  <!-- Optional Debug Parameters -->
  <arg name="debug" default="false"/>

  <!-- Point cloud topic argument -->
  <arg name="point_cloud_topic" default="/points_filtered"/>

  <!-- Camera optical frame argument -->
  <arg name="camera_optical_frame" default="$(arg namespace)/camera_depth_optical_frame"/>

  <!-- Process Tracker With Cloud Node -->
  <node name="process_tracker_with_cloud" pkg="auv_mapping" type="process_tracker_with_cloud" output="screen">
    <!-- Parameters -->
    <param name="camera_info_topic" value="/$(arg namespace)/camera/color/camera_info"/>
    <param name="lidar_topic" value="$(arg point_cloud_topic)"/>
    <param name="yolo_result_topic" value="/yolo_result_realsense"/>
    <param name="camera_optical_frame" value="$(arg camera_optical_frame)"/>
    <param name="cluster_tolerance" value="0.2"/>
    <param name="voxel_leaf_size" value="0.05"/>
    <param name="min_cluster_size" value="50"/>
    <param name="max_cluster_size" value="10000"/>
    <param name="roi_expansion_factor" value="0.90"/>
    <!-- Remaps -->
    <remap from="update_object_transforms" to="/$(arg namespace)/map/object_transform_updates"/>
  </node>

  <!-- Start RViz Optionally-->
  <node if="$(arg debug)" pkg="rviz" type="rviz" name="rviz"/>
</launch>
